File: QuanLyPhongKham.Server/src/QuanLyPhongKham.DTOs/Responses/DonThuocResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuanLyPhongKham.DTOs.Responses
{
    public class DonThuocResponse
    {
        public Guid MaDT { get; set; }
        public Guid? MaPK { get; set; }
        public DateTime NgayKe { get; set; }
        public List<ChiTietDonThuocResponse> ChiTietDonThuocs { get; set; } = [];
        public decimal TongTien => ChiTietDonThuocs.Sum(x => x.ThanhTien);
    }
}

File: QuanLyPhongKham.Client/src/interfaces/responses/DonThuocResponse.js
/**
 * @typedef {Object} DonThuocResponse
 * @property {string} maDT
 * @property {string} maPK
 * @property {string|Date} ngayKe
 * @property {Array} chiTietDonThuocs
 */

export const createDonThuocResponse = (
  maDT = "",
  maPK = "",
  ngayKe = null,
  chiTietDonThuocs = [],
) => ({
  maDT,
  maPK,
  ngayKe,
  chiTietDonThuocs,
});
